homeassistant/components/smhi/config_flow.py

```diff
@@ -48,6 +48,10 @@ async def async_step_user(
             if is_ok:
                 name = slugify(user_input[CONF_NAME])
                 if not self._name_in_configuration_exists(name):
+                    latitude = user_input[CONF_LATITUDE]
+                    longitude = user_input[CONF_LONGITUDE]
+                    await self.async_set_unique_id(f"{latitude}-{longitude}")
+                    self._abort_if_unique_id_configured()
                     return self.async_create_entry(
                         title=user_input[CONF_NAME], data=user_input
```

Comment:
Side note: Would be nice to not ask the user for a name at all and just use the coordinates or similar available data as default title.

Reply:
The name is also used on the sensors and surely I would prefer myself if the sensors is named "Västerås rain" instead of something with the coordinates. So not seeing much options.

Reply:
Yes, a weather platform can set DeviceInfo and set the entry type to service. See other weather integrations for example.

Just set a default name as the name. It doesn't have to be the coordinates either. The user can rename the config entry, device and entities to whatever they want.

Reply:
Ok, I will do that in a follow-up PR given it's not really in scope for this one.